barretenberg/cpp/src/barretenberg/honk/sumcheck/sumcheck.hpp

```diff
@@ -206,6 +206,7 @@ template <typename Flavor> class SumcheckVerifier {
             round.compute_next_target_sum(round_univariate, round_challenge);
             pow_univariate.partially_evaluate(round_challenge);
 
+            // TODO(#726): Properly handle this in the recursive setting.
```

Comment:
Inside the circuit there should be no if at all in this particular case, right?

Reply:
Yeah the problem is right now this code is shared by recursive and native verification so some choices need to be made. I'll handle this in a follow on for the issue referenced